packages/bierzo-wallet/src/logic/connection.ts

```diff
@@ -1,34 +1,42 @@
 import { BlockchainConnection, ChainId } from "@iov/bcp";
-import { BnsConnection } from "@iov/bns";
+import { BnsConnection, createBnsConnector } from "@iov/bns";
 import { EthereumConnection } from "@iov/ethereum";
-import { LiskConnection } from "@iov/lisk";
+import { createEthereumConnector, EthereumConnectionOptions } from "@iov/ethereum";
+import { createLiskConnector, LiskConnection } from "@iov/lisk";
 
 import { ChainSpec, getConfig } from "../config";
 
 let ethereumConnection: EthereumConnection | undefined;
 let bnsConnection: BnsConnection | undefined;
 let liskConnection: LiskConnection | undefined;
 
-async function getEthereumConnection(url: string, scraperApiUrl?: string): Promise<EthereumConnection> {
+async function getEthereumConnection(
+  url: string,
+  chainId: ChainId,
+  options: EthereumConnectionOptions,
+): Promise<EthereumConnection> {
   if (!ethereumConnection) {
+    const connector = createEthereumConnector(url, options, chainId);
     // eslint-disable-next-line require-atomic-updates
-    ethereumConnection = await EthereumConnection.establish(url, { scraperApiUrl });
+    ethereumConnection = (await connector.establishConnection()) as EthereumConnection;
   }
   return ethereumConnection;
 }
 
-async function getBnsConnection(url: string): Promise<BnsConnection> {
+async function getBnsConnection(url: string, chainId: ChainId): Promise<BnsConnection> {
   if (!bnsConnection) {
+    const connector = createBnsConnector(url, chainId);
     // eslint-disable-next-line require-atomic-updates
-    bnsConnection = await BnsConnection.establish(url);
+    bnsConnection = (await connector.establishConnection()) as BnsConnection;
   }
   return bnsConnection;
 }
 
-async function getLiskConnection(url: string): Promise<LiskConnection> {
+async function getLiskConnection(url: string, chainId: ChainId): Promise<LiskConnection> {
   if (!liskConnection) {
+    const connector = createLiskConnector(url, chainId);
     // eslint-disable-next-line require-atomic-updates
-    liskConnection = await LiskConnection.establish(url);
+    liskConnection = (await connector.establishConnection()) as LiskConnection;
```

Comment:
I think we can change the return types of all `getLiskConnection`, `getBnsConnection`, `getEthereumConnection` as well as the 3 variables in this file to `BlockchainConnection`. This avoids the needs for all the casting

Reply:
Yeah, the real issue is that we don't get the specific types from IOV-Core as we should. I created a ticket for that. In the case of this file, we don't need the specific types.